getting-started/building-your-implementation-plan.md

```diff
@@ -0,0 +1,95 @@
+# Building your Implementation Plan
+
+Nosto provides plugins for the most common eCommerce platforms like Shopify, Magento 2, Shopware 6, BigCommerce and Prestashop.
+
+- The plugins have implemented the Nosto backend APIs so in most cases 90% of the product catalog sync is already handled.
+  - If your client has custom use cases, you likely will have to extend the Nosto plugin, change the data structure within your platform or make use of one of the Nosto backend APIs to ensure all the needed product data is in Nosto.
+- Depending on your platform, a Nosto plugin (like for Shopware or Magento 2) extends the default theme (like Dawn, Storefront, Luma or Hyvä) and implements the frontend mechanisms that are explained in the following.
+  - Some platform plugins only include a very basic frontend component like adding the Nosto script and the page tagging.
+  - If your client has a custom frontend (running headless or a SPA*), you will need to implement the frontend mechanisms one by one.
+
+*This guide focuses on classic web applications that work with full page reloads. For SPAs/PWAs like React, the same principles apply but they are handled differently.
+We encourage you to first read through the guiding principles and in a next step make yourself familiar with the implementation within a Single Page Application.
+
+
+## Components of a stable Nosto Implementation
+
+1. Sending/updating the Product Catalog via GraphQL API (or REST API) to Nosto regularly.
+   - For custom platforms (without Nosto plugin), you must use one of the APIs and build the product sync yourself (no CSV or feed option). If you’re working with PHP, you can use the Nosto PHP-SDK.
+2. Page tagging via JavaScript or hidden HTML on all pages
+   - There are several page types, so please make sure they are covered in your QA checklist. (You can use ours as a blueprint.)
+   - The current cart content and logged in user information need to be available on all pages.
+3. Placements (empty divs) where Nosto content can be injected.
+   - These placements should be added into the HTML, most Nosto plugins offer CMS blocks for easier use.
+4. Injecting personalized content into the placements of the current page (banners, product recommendation carousels/grids, ...)
+   - This involves a simple request/response pattern: Tell Nosto which placements are on the current page and you’ll receive the content as JSON or HTML.
+5. Replacing native platform features like product listings (PLPs and SERPs) with facets, pagination and sorting options.
```

Comment:
Lots of merchants replace platform features with Nosto features without considering the case of Nosto failing, so there is lots of noise when there is an issue with Nosto, because nothing works anymore. Perhaps this is already a good place to highlight the importance of falling back to platform features when the corresponding Nosto feature returns an error or times out.